src/Kestrel.Transport.Sockets/SocketTransport.cs

```diff
@@ -89,14 +94,22 @@ public Task BindAsync()
         {
             if (_listenSocket != null)
             {
-                var listenSocket = _listenSocket;
-                _listenSocket = null;
-
-                listenSocket.Dispose();
+                _unbinding = true;
+                _listenSocket.Dispose();
 
                 Debug.Assert(_listenTask != null);
                 await _listenTask.ConfigureAwait(false);
+
+                _unbinding = false;
+                _listenSocket = null;
                 _listenTask = null;
+
+                if (_listenException != null)
+                {
+                    var exInfo = ExceptionDispatchInfo.Capture(_listenException);
```

Comment:
What's the point of capturing and then immediately rethrowing? Aren't you supposed to `.Capture` at the point where the exception occurs and then `.Throw`?

I may be misinterpreting how EDI works though.

Reply:
You don't have to capture at the point of exception. See https://github.com/dotnet/corefx/pull/17570

Reply:
I withdraw my concerns then :). Thanks for clarifying.